### ISSUE TITLE
Auto-resizing only working in one direction

### ISSUE THREAD

**Issue by Brideau** [twitter/d3kit#14]
I'm working with a chart using `autoResize` to `true` and `autoResizeToAspectRatio` set to 1.5. I noticed that if I shrunk the browser window, the chart resized properly, but if I then made it large again, the chart stayed small.

The issue seems to come from `resizeToAspectRatio` function. For example, if you start with a large div and go to a small one, the div proportions go from:
- _totalWidth = 700
- _totalHeight = 467
- estimatedHeight = 467

To:
- _totalWidth = 400
- _totalHeight = 467
- estimatedHeight = 267

The if/else statement on [line 663](https://github.com/twitter/d3kit/blob/master/dist/d3kit.js#L663) then correctly says "if the estimated height is not larger than the height, change the height to the estimated height."

In the opposite direction, however, it does this:
- _totalWidth = 400
- _totalHeight = 267
- estimatedHeight = 267

To:
- _totalWidth = 700
- _totalHeight = 267
- estimatedHeight = 467

The same if/else statement then effectively says "if the estimated height is greater than the total height, adjust the width." But the width is already adjusted, so nothing updates.

I patched this by changing:

```
  if (estimatedH > h) {
    width(Math.floor(h * ratio), doNotDispatch);
  } else {
    height(estimatedH, doNotDispatch);
  }
```

To:

```
  if (estimatedH > h) {
    height(estimatedH, doNotDispatch);
    width(Math.floor(estimatedH * ratio), doNotDispatch);
  } else {
    height(estimatedH, doNotDispatch);
  }
```

But, I'm not sure if that works in all cases, so perhaps there's a cleaner solution.

**Comment by kristw** [comment]
I will have to take a closer look at this. We were thinking about refactoring the whole resizing logic in #12 . Will update once we have any progress. Meanwhile, I am glad that you have found a workaround.

**Comment by kristw** [comment]
This issue was fixed in d3kit v3.1.1.
